sdk/go/pulumi/run.go

```diff
@@ -38,15 +38,22 @@ type RunOption func(*RunInfo)
 // to register resources and orchestrate deployment activities.  This connects back to the Pulumi engine using gRPC.
 // If the program fails, the process will be terminated and the function will not return.
 func Run(body RunFunc, opts ...RunOption) {
-	if err := RunErr(body, opts...); err != nil {
-		if err != ErrPlugins {
-			fmt.Fprintf(os.Stderr, "error: program failed: %v\n", err)
-			os.Exit(1)
-		}
+	err := RunErr(body, opts...)
```

Comment:
So I'm reading this diff here and the only thing that really changed it seems like was substituting

```
fmt.Fprintf(os.Stderr, "error: program failed: %v\n", err)
```

with

```
ctx.Log.Error(fmt.Sprintf("an unhandled error occurred: program failed: \n%v", err), nil)
```

This change makes sense!

As to reordering if branches, it's nitty/style thing but from the sense of style in Go I've developed so far, the code was better before the change. We use this style all the time:

```
if <error-cond> {
   <handle error>
}
if <error-cond> {
   <handle error>
}
<handle happy path>
``` 

The new style here is

```
if <no err> {
    <handle happy path>
}
<handle error>
```

Seems a bit weird.

Reply:
Glad to be able to generate discussion around code style.

The current code looks like this
```
if <stmt>; <error> {
    if <special error> {
        <handle error>
        exit()
    }
    <handle error>
    exit()
}
<happy path>
```

The original code has no code in its happy path except `<stmt>`, but that's indented into the if statement so the conditions where the branch runs is confusing.

All the code for Run is error handling as it outsources the logic to `RunErr`

So it looks like `err != ErrPlugins` could possibly be `nil` but it's already guarded by the parent if statement so it doesn't exit at this location so the error handling logic would end up reading out of order since the error handling is at the bottom.